ga4mp/ga4mp.py

```diff
@@ -57,12 +64,34 @@ class BaseGa4mp(object):
     """
 
     def __init__(self, api_secret):
+        self._initialization_time = time.time()
         self.api_secret = api_secret
         self._event_list = []
-        self._user_properties = {}
+        # Because it's required, initialize a starter store that the user can overwrite if desired.
+        self.create_store()
         self._base_domain = "https://www.google-analytics.com/mp/collect"
         self._validation_domain = "https://www.google-analytics.com/debug/mp/collect"
 
+    def create_store(self, use_file=False, store=None, session_id=None, data_location=None):
```

Comment:
This ties your store implementation to either DictStore or FileStore. If I write my own I cannot plug it in without changing the core code of this base class.
Store needs to come from the outside into the BaseGa4mp class so that it is not tightly coupled to the two possible implementations you have now.

It would be something like:
```
self.__init__(self, api_secret, store: BaseStore = None):
    assert store is None or isinstance(store, BaseStore)
    self._store = store if store else DictStore()
```

Reply:
@zvizdo That makes sense, but it creates a new problem that I'm not sure how to address, which is how to instruct the user on creating a store if they want to use their own.

In their code, would it be something like:
```py
from ga4mp import GtagMP
from ga4mp.store import FileStore

new_store = FileStore(data_location=".folder/file.json")

tracker = GtagMP(api_secret="934TXS", measurement_id="G-12345", client_id="1234852.1235081235", store=new_store)
```

Is there a less clunky way for them to import DictStore or FileStore?

And should I then remove FileStore from the imports in ga4mp.py?

Reply:
I would make it so that if the store is not passed to the client (GtagMP) then internally it defaults to DictStore and the user will never know, and also won't see any difference.

If they want to use a special store (any that is not DictStore) they would need to make an instance of it ahead of time, exactly how you have it above.

Reply:
Update: after talking with about this with another Python developer I know, I've learned that it's _fine_ to have a deeper import like the one described. Marking this resolved, and I'm going to start updating the README.